### ISSUE TITLE
Lint for returning a pointer to stack memory associated with a local variable

### ISSUE THREAD

**Issue by 1c3t3a** [rust-lang/rust-clippy#13808]
### What it does

This adds a new lint that warns about returning a pointer to stack memory associated with a local variable. This pointer will always be invalid.

### Advantage

- It warns about code that follows a pattern that is very unlikely to be correct.

### Example

```rust
pub fn foo() -> *const i32 {
    let x = 42;
    &x
}
```

Should not be written that way.

**Comment by 1c3t3a** [comment]
I would propose to make this a [`warn-by-default`](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html) rustc lint. I am not sure if that is the right approach and whether Clippy is the right place to open that bug, I am happy for guidance!

**Comment by 1c3t3a** [comment]
Closing this in favor of the [rust repository issue](https://github.com/rust-lang/rust/issues/134215).
